wcfsetup/install/files/js/WoltLabSuite/Core/Acp/Ui/Package/PrepareInstallation.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Attempts to download the requested package from the file and prompts for the
+ * authentication credentials on rejection.
+ * 
+ * @author      Alexander Ebert
+ * @copyright   2001-2019 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module      WoltLabSuite/Core/Acp/Ui/Package/PrepareInstallation
+ */
+define(['Ajax', 'Core', 'Language', 'Ui/Dialog'], function(Ajax, Core, Language, UiDialog) {
+	'use strict';
+	
+	function AcpUiPackagePrepareInstallation() { }
+	AcpUiPackagePrepareInstallation.prototype = {
+		/**
+		 * @param {string} identifier
+		 * @param {string} version
+		 */
+		start: function (identifier, version) {
+			this._identifier = identifier;
+			this._version = version;
+			
+			this._prepare({});
+		},
+		
+		/**
+		 * @param {Object} authData
+		 */
+		_prepare: function(authData) {
+			var packages = {};
+			packages[this._identifier] = this._version;
+			
+			Ajax.api(this, {
+				parameters: {
+					authData: authData,
+					packages: packages
+				}
+			});
+		},
+		
+		/**
+		 * @param {number} packageUpdateServerId
+		 * @param {Event} event
+		 */
+		_submit: function(packageUpdateServerId, event) {
+			event.preventDefault();
+			
+			var usernameInput = elById('packageUpdateServerUsername');
+			var passwordInput = elById('packageUpdateServerPassword');
+			
+			elInnerError(usernameInput, false);
+			elInnerError(passwordInput, false);
+			
+			var username = usernameInput.value.trim();
+			if (username === '') {
+				elInnerError(usernameInput, Language.get('wcf.global.form.error.empty'));
+			}
+			else {
+				var password = passwordInput.value.trim();
+				if (password === '') {
+					elInnerError(passwordInput, Language.get('wcf.global.form.error.empty'));
+				}
+				else {
+					this._prepare({
+						packageUpdateServerID: packageUpdateServerId,
+						password: password,
+						saveCredentials: elById('packageUpdateServerSaveCredentials').checked,
+						username: username
+					});
+				}
+			}
+		},
+		
+		_ajaxSuccess: function(data) {
+			if (data.returnValues.queueID) {
+				UiDialog.close(this);
```

Comment:
When I trie to install a package, I get the following error: 
```
Uncaught Error: Expected a valid dialog id, '[object Object]' does not match any active dialog.
    at Object.close (Dialog.js?t=1550865154:792)
    at AcpUiPackagePrepareInstallation._ajaxSuccess (PrepareInstallation.js?t=1550865154:76)
    at AjaxRequest._success (Request.js?t=1550865154:257)
    at XMLHttpRequest._xhr.onload (Request.js?t=1550865154:135)
```

Reply:
Thanks, fixed.